libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/Evaluate.hs

```diff
@@ -218,33 +213,6 @@ data PlutusDebugInfo c
       (Maybe P.ExBudget)
   deriving (Show)
 
-debugPlutus :: Crypto c => String -> IO (PlutusDebugInfo c)
```

Comment:
Unfortunately, we cannot remove this function from the library, because it looks like there are already downstream users of it. In particular I found that hydra uses this functionality.

What I suggest in order to resolve it is to create a type in this module:
```haskell
data PlutusDebugOverrides = PlutusDebugOverrides
  { pdoScript :: !(Maybe ByteString)
  , pdoProtocolVersion :: !(Maybe Version)
  , pdoLanguage :: !(Maybe Language)
  , pdoCostModelValues :: !(Maybe [Int64])
  , pdoExUnitsMem :: !(Maybe Natural)
  , pdoExUnitsSteps :: !(Maybe Natural)
  }
```

then the `Opts` in CLI become:

```haskell
data Opts = Opts
  { optsScriptWithContext :: !String
  , optsOverrides :: !PlutusDebugOverrides
  }
  deriving (Show)
```

Then the only breaking change to this function would be addition of a new argument:
```haskell
debugPlutus :: Crypto c => String -> PlutusDebugOverrides -> IO (PlutusDebugInfo c)
```